hbase-client/src/main/java/org/apache/hadoop/hbase/client/Admin.java

```diff
@@ -2651,4 +2651,7 @@ List<LogEntry> getLogEntries(Set<ServerName> serverNames, String logType, Server
    * Get the list of cached files
    */
   List<String> getCachedFilesList(ServerName serverName) throws IOException;
+
+  @InterfaceAudience.Private
+  void restoreBackupSystemTable(String snapshotName) throws IOException;
```

Comment:
If this doesn't need to be public, it shouldn't go on the very public Admin interface. You could use `Admin#execProcedure()` instead because you're basically just starting a procedure. You could also add an PRC to `MasterService` that you could call from `BackupSystemTable`. These would both accomplish communicating to the HMaster, but more privately.

Reply:
execProcedure only works for distributed procedures, which require cluster-wide coordination, so I didn't think that it applied to this case, plus it makes the table locking stuff more difficult. 

> You could also add an PRC to MasterService that you could call from BackupSystemTable

Isn't that what the Admin is doing in this case? It's connecting to the MasterRpcServices. Otherwise, I need to create an entirely new type of connection wrapper class in order to interface with the MasterRPCServices. I'm happy to do that if you think that's a better approach. 

Let me know if I'm misunderstanding something though

Reply:
> Otherwise, I need to create an entirely new type of connection wrapper class in order to interface with the MasterRPCServices. I'm happy to do that if you think that's a better approach.

Yes, that's what I'm suggesting. I think it's gonna be pretty lightweight. If you look at `HBaseHbck` that makes it look pretty easy.

Reply:
Yeah, fair enough.